### PR TITLE
Remove the IBuildOnlyDiagnosticService

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@mavasani ptal

**Comment by mavasani** [comment]
@CyrusNajmabadi are we planning to have a replacement for this in the editor side? This service was added to enable Suppression code fixes for build only diagnostics, something we got repeated customer requests for. If not from the lightbulb, we may want to at least enable these fixes from the error list. Something to consider in the future.

**Review by mavasani — APPROVED** [review]

**Comment by mavasani** [comment]
> @CyrusNajmabadi are we planning to have a replacement for this in the editor side? This service was added to enable Suppression code fixes for build only diagnostics in the lightbulb, something we got repeated customer requests for. If not from the lightbulb, we may want to at least enable these fixes from the error list. Something to consider in the future.

Tagging @genlu - wondering if there is scope to enable generating these suppression code fixes in the error list with use of Copilot?

**Comment by CyrusNajmabadi** [comment]
hrmm... i'll leave this in for now then.

File: src/Features/LanguageServer/Protocol/Features/CodeFixes/CodeFixService.cs [hunk 1]
```diff
@@ -205,10 +202,7 @@ internal partial class CodeFixService : ICodeFixService
             var copilotDiagnostics = await GetCopilotDiagnosticsAsync(document, range, priorityProvider.Priority, cancellationToken).ConfigureAwait(false);
             diagnostics = diagnostics.AddRange(copilotDiagnostics);
 
-            var buildOnlyDiagnosticsService = document.Project.Solution.Services.GetRequiredService<IBuildOnlyDiagnosticsService>();
-            var buildOnlyDiagnostics = buildOnlyDiagnosticsService.GetBuildOnlyDiagnostics(document.Id);
```
Comment (CyrusNajmabadi) [inline src/Features/LanguageServer/Protocol/Features/CodeFixes/CodeFixService.cs:209]: moving to draft as i'm not yet comfportable with the change in behavior here.

**Comment by CyrusNajmabadi** [comment]
This service is gone.

**Pull request closed without merging**